### PR TITLE
Intake and containment statemachines

### PR DESCRIPTION
**Pull request by Matthew13479**

### CODE REVIEW

File: pyswervedrive/swervechassis.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def execute(self):
 
         pid_z = 0
         if self.hold_heading:
-            if self.momentum and abs(self.imu.getHeadingRate()) < 0.005:
+            if self.momentum and abs(self.imu.getHeadingRate()) < 0.0001:
```
Comment (auscompgeek) [inline pyswervedrive/swervechassis.py]: This hack should not be in your code…

File: components/intake.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ def cube_inside(self):
 
     def contacting_cube(self):
         """Returns True of the current output of the motor is above 3"""
```
Comment (auscompgeek) [inline components/intake.py]: Please fix your documentation.

**Review by auscompgeek — CHANGES_REQUESTED** [review]

File: components/intake.py [hunk 3]
```diff
@@ -72,18 +74,16 @@ def extension(self, value):
 
     def infrared_distance(self):
         """Gets the distance of the infrared sensor in mm"""
```
Comment (auscompgeek) [inline components/intake.py]: I think this documented behaviour is wrong.

File: robot.py [hunk 4]
```diff
@@ -81,6 +81,8 @@ def createObjects(self):
         self.joystick = wpilib.Joystick(0)
         self.gamepad = wpilib.XboxController(1)
 
+        self.sd = NetworkTables.getTable("SmartDashboard")
```
Comment (auscompgeek) [inline robot.py]: …

File: robot.py [hunk 5]
```diff
@@ -68,6 +67,7 @@ def createObjects(self):
         self.intake_kicker = wpilib.Solenoid(1)
         self.extension_arms = wpilib.Solenoid(3)
         self.infrared = SharpIRGP2Y0A41SK0F(0)
+        self.lift_motor = ctre.WPI_TalonSRX(0)
```
Comment (auscompgeek) [inline robot.py]: ????????

**Review by fizzlebert — APPROVED** [review]

**Pull request merged**